tests/e2e/api/general.test.ts

```diff
@@ -108,4 +95,120 @@ describe("general api", () => {
+
+    test("it fetches view function with a type", async () => {
+      const config = new AptosConfig({ network: Network.LOCAL });
+      const aptos = new Aptos(config);
+
+      const payload: InputViewRequestData = {
+        function: "0x1::chain_id::get",
+      };
+
+      const chainId = (await aptos.view<[number]>({ payload }))[0];
+
+      expect(chainId).toEqual(4);
+    });
+
+    test("it fetches view function with bool", async () => {
+      const config = new AptosConfig({ network: Network.LOCAL });
+      const aptos = new Aptos(config);
+
+      const payload: InputViewRequestData = {
+        function: "0x1::account::exists_at",
+        functionArguments: ["0x1"],
+      };
+
+      const exists = (await aptos.view<[boolean]>({ payload }))[0];
+
+      expect(exists).toBe(true);
+
+      const payload2: InputViewRequestData = {
+        function: "0x1::account::exists_at",
+        functionArguments: ["0x12345"],
+      };
+
+      const exists2 = (await aptos.view<[boolean]>({ payload: payload2 }))[0];
+
+      expect(exists2).toBe(false);
+    });
+
+    test("it fetches view function with address input and different output types", async () => {
+      const config = new AptosConfig({ network: Network.LOCAL });
+      const aptos = new Aptos(config);
+
+      const payload: InputViewRequestData = {
+        function: "0x1::account::get_sequence_number",
+        functionArguments: ["0x1"],
+      };
+
+      const sequenceNumber = (await aptos.view<[string]>({ payload }))[0];
+
+      expect(BigInt(sequenceNumber)).toEqual(BigInt(0));
+
+      const payload2: InputViewRequestData = {
+        function: "0x1::account::get_authentication_key",
+        functionArguments: ["0x1"],
+      };
+
+      const authKey = (await aptos.view<[string]>({ payload: payload2 }))[0];
+
+      expect(authKey).toEqual("0x0000000000000000000000000000000000000000000000000000000000000001");
+    });
+
+    test("it fetches view functions with generics", async () => {
+      const config = new AptosConfig({ network: Network.LOCAL });
+      const aptos = new Aptos(config);
+
+      const payload: InputViewRequestData = {
+        function: "0x1::coin::symbol",
+        typeArguments: ["0x1::aptos_coin::AptosCoin"],
+      };
+
+      const symbol = (await aptos.view<[string]>({ payload }))[0];
+      expect(symbol).toEqual("APT");
+
+      const payload2: InputViewRequestData = {
+        function: "0x1::coin::is_account_registered",
+        typeArguments: ["0x1::aptos_coin::AptosCoin"],
+        functionArguments: ["0x1"],
+      };
+
+      const isRegistered = (await aptos.view<[boolean]>({ payload: payload2 }))[0];
+      expect(isRegistered).toEqual(false);
+
+      const payload3: InputViewRequestData = {
+        function: "0x1::coin::supply",
+        typeArguments: ["0x1::aptos_coin::AptosCoin"],
+        functionArguments: [],
+      };
+
+      const supply = (await aptos.view<[{ vec: [string] }]>({ payload: payload3 }))[0].vec[0];
```

Comment:
so painful :P

I also just realized this morning while writing tests for view function outputs is that the returned results are serialized in some cases. It seems like the inputs *and* the outputs are a frankenstein of a JSON mapping and BCS-serialized hex strings...

Reply:
I suspect with improvements on the actual API, a lot of this can be improved upon / cleaned up.